cognite/client/data_classes/shared.py

```diff
@@ -47,3 +47,94 @@ def __init__(self, count: int = None, value: Union[int, str] = None, **kwargs):
         self.value = value
 
     value = CognitePropertyClassUtil.declare_property("value")
+
+
+class Geometry(dict):
+    """Represents the points, curves and surfaces in the coordinate space.
+
+    Args: type (str): The geometry type. One of 'Point', 'MultiPoint, 'LineString', 'MultiLineString', 'Polygon', or 'MultiPolygon'.
+          coordinates (List): An array of the coordinates of the geometry. The structure of the elements in this array is determined by the type of geometry.
+    """
+
+    def __init__(self, type: str, coordinates: List):
+        self.type = type
+        self.coordinates = coordinates
+
+    type = CognitePropertyClassUtil.declare_property("type")
+    coordinates = CognitePropertyClassUtil.declare_property("coordinates")
+
+    @classmethod
+    def _load(self, raw_geometry: Dict[str, Any]):
+        return Geometry(type=raw_geometry["type"], coordinates=raw_geometry["coordinates"])
+
+    def dump(self, camel_case: bool = False):
+        dump_key = lambda key: key if not camel_case else utils._auxiliary.to_camel_case(key)
+        return {dump_key(key): value for key, value in self.items()}
+
+
+class GeometryFilter(dict):
+    """Represents the points, curves and surfaces in the coordinate space.
+
+    Args: type (str): The geometry type. One of 'Point', 'LineString', 'MultiLineString', 'Polygon', or 'MultiPolygon'.
+          coordinates (List): An array of the coordinates of the geometry. The structure of the elements in this array is determined by the type of geometry.
+    """
+
+    def __init__(self, type: str, coordinates: List):
+        self.type = type
+        self.coordinates = coordinates
+
+    type = CognitePropertyClassUtil.declare_property("type")
+    coordinates = CognitePropertyClassUtil.declare_property("coordinates")
+
+
+class GeoLocation(dict):
+    """A GeoLocation object conforming to the GeoJSON spec.
+
+    Args: type (str): The GeoJSON type. Currently only 'Feature' is supported.
+          geometry (object): The geometry type. One of 'Point', 'MultiPoint, 'LineString', 'MultiLineString', 'Polygon', or 'MultiPolygon'.
+          properties (object): Additional properties in a String key -> Object value format.
```

Comment:
Is this a metadata associated to files?

Reply:
Similar to metadata, but it's related to the `geoLocation`. Optional though, I'll clarify that.